src/pandapipes/component_models/component_toolbox.py

```diff
@@ -15,6 +15,18 @@
 from pandapipes.pf.internals_toolbox import _sum_by_group
 
 
+def get_internal_node_lookup_structure(net, internals, table_name, internal_nodes,
+                                       internal_nodes_number, node_start, node_end):
+    internals[table_name] = np.empty((internal_nodes_number, 2), dtype=np.int32)
+    internals[table_name][:, 0] = np.repeat(net[table_name].index, internal_nodes.astype(np.int32))
+    internals[table_name][:, 1] = np.arange(node_start, node_end)
```

Comment:
Should we maybe add another column that gives the internal number of the node on the branch? It could be adjusted by the branch element itself, but is by default just the cumsum for the respective branch index.